cloud/docker/docker_image.py

```diff
@@ -1,457 +1,502 @@
+    default: false
+  pull:
     description:
-      - Path to the PEM-encoded key used to authenticate docker client. If
-        specified tls_client_cert must be valid
-    default: ${DOCKER_CERT_PATH}/key.pem
-    version_added: "2.0"
-  tls_ca_cert:
+      - When building an image downloads any updates to the FROM image in Dockerfiles.
+    default: true
+  rm:
     description:
-      - Path to a PEM-encoded certificate authority to secure the Docker connection.
-        This has no effect if use_tls is encrypt.
-    default: ${DOCKER_CERT_PATH}/ca.pem
-    version_added: "2.0"
-  tls_hostname:
+      - Remove intermediate containers after build.
+    default: true
+  nocache:
     description:
-      - A hostname to check matches what's supplied in the docker server's
-        certificate.  If unspecified, the hostname is taken from the docker_url.
-    default: Taken from docker_url
-    version_added: "2.0"
-  docker_api_version:
+      - Do not use cache when building an image.
+    default: false
+  repository:
     description:
-      - Remote API version to use. This defaults to the current default as
-        specified by docker-py.
-    default: docker-py default remote API version
-    version_added: "2.0"
+      - Full path to a repository. Use with state 'present' to tag the image into the repository.
+    required: false
+    default: null
   state:
     description:
-      - Set the state of the image
-    required: false
+      - Make assertions about the state of an image.
+      - When 'absent' an image will be removed. Use the force option to un-tag and remove all images
+        matching the provided name.
+      - When 'present' check if an image exists using the provided name and tag. If the image is not found or the
+        force option is used, the image will either be pulled, built or loaded. By default the image will be pulled
+        from Docker Hub. To build the image, provide a path value set to a directory containing a context and
+        Dockerfile. To load an image, specify load_path to provide a path to an archive file. To tag an image to a
+        repository, provide a repository path.
+      - "NOTE: 'build' is DEPRECATED. Specifying 'build' will behave the same as 'present'."
     default: present
-    choices: [ "present", "absent", "build" ]
-    aliases: []
-  timeout:
+    choices:
+      - absent
+      - present
+      - build (DEPRECATED)
+  tag:
     description:
-      - Set image operation timeout
-    required: false
-    default: 600
-    aliases: []
-requirements:
-    - "python >= 2.6"
-    - "docker-py"
-    - "requests"
-'''
-
-EXAMPLES = '''
-Build docker image if required. Path should contains Dockerfile to build image:
+      - Used to select an image when pulling. Will be added to the image when pushing, tagging or building. Defaults to
+       'latest' when pulling an image.
+    default: latest
+  use_tls:
```

Comment:
Not sure what you mean. The tls and tls_verify parameters are handled in the common module. use_tls is here just to be compatible with the old docker_image. 

It's a bit late to change tls and tls_verify now, if that's what you're suggesting. They match exactly what docker-py has, and they match the environment vars (DOCKER_TLS and DOCLER_TLS_VERIFY) used by Docker machine. They are confusing, yes. But, it's consistent with how docker does things.